src/material.h

```diff
@@ -75,4 +77,13 @@ class material_type
         const mat_burn_data &burn_data( size_t intensity ) const;
 };
 
+namespace materials
```

Comment:
> Imagine you have more than one container for the objects

Additional functionality. No known current uses?

> hence no additional cost

Different interface to existing code

> YAGNI is barely relevant here

You don't need the additional functionality as part of this PR. If you can give a specific example of where it might be imminently used then that's a valid rebuttal.